src/main/java/de/komoot/photon/PhotonDoc.java

```diff
@@ -173,31 +149,54 @@ public void completeFromAddress() {
     /**
      * Extract an address field from an address tag and replace the appropriate address field in the document.
      *
-     * @param existingField The current value of the document's address field.
+     * @param addressType The type of address field to fill.
      * @param addressFieldName The name of the address tag to use (without the 'addr:' prefix).
      *
      * @return 'existingField' potentially with the name field replaced. If existingField was null and
      *         the address field could be found, then a new map with the address as single entry is returned.
      */
-    private Map<String, String> extractAddress(Map<String, String> existingField, String addressFieldName) {
+    private void extractAddress(AddressType addressType, String addressFieldName) {
```

Comment:
Could you get addressFieldName also from addressType.getName? Then you could remove it from the parameter list.

BTW: thanks for introducing me to EnumMap - really interesting :)

Reply:
The mapping between the GeocodeJSON names and the OSM ` addr:*`  tag names is not exactly 1:1. suburb and neighbourhood are already different and we might even add more variants (like 'province'  for 'state'). We could add the mapping into the new AddressType enum as an extra field though. Something for an extra PR.